featurebyte/query_graph/node/generic.py

```diff
@@ -1418,22 +1463,55 @@ def derive_var_type(self, inputs: List[OperationStructure]) -> DBVarType:
         var_name_generator: VariableNameGenerator,
         operation_structure: OperationStructure,
         config: CodeGenerationConfig,
-    ) -> Tuple[List[StatementT], VarNameExpressionStr]:
-        var_name_expr = input_var_name_expressions[0]
-        mask_var_name_expr = input_var_name_expressions[1]
+        context: CodeGenerationContext,
+    ) -> Tuple[List[StatementT], VarNameExpressionInfoStr]:
+        var_name_expressions = self._assert_no_info_str(node_inputs)
+        var_name_expr = var_name_expressions[0]
+        mask_var_name_expr = var_name_expressions[1]
         statements, var_name = self._convert_expression_to_variable(
             var_name_expression=var_name_expr,
             var_name_generator=var_name_generator,
             node_output_type=operation_structure.output_type,
             node_output_category=operation_structure.output_category,
+            to_associate_with_node_name=False,
         )
+        mask_statements, mask_var_name = self._convert_expression_to_variable(
+            var_name_expression=mask_var_name_expr,
+            var_name_generator=var_name_generator,
+            node_output_type=NodeOutputType.SERIES,
+            node_output_category=operation_structure.output_category,
+            to_associate_with_node_name=False,
+            variable_name_prefix="mask",
+        )
+        statements.extend(mask_statements)
+        # only make the copy if we are not generating info string
+        # if generating info string, we will delay the copy to the assign node
+        var_statements, output_var_name = self._convert_to_proper_variable_name(
+            var_name=var_name,
+            var_name_generator=var_name_generator,
+            operation_structure=operation_structure,
+            required_copy=context.required_copy and not context.as_info_str,
+            to_associate_with_node_name=True,
+        )
+        statements.extend(var_statements)
+
         value: RightHandSide = ValueStr.create(self.parameters.value)
-        if len(input_var_name_expressions) == 3:
-            value = input_var_name_expressions[2]
-        statements.append((VariableNameStr(f"{var_name}[{mask_var_name_expr}]"), value))
-        return statements, var_name
+        if len(var_name_expressions) == 3:
+            value = var_name_expressions[2]
+
+        if context.as_info_str:
+            # This is to handle the case where `View[<column>][<condition>] = <value>` is used.
+            # Since there is no single line in SDK code to generate conditional expression, we output info instead
+            # and delay the generation of SDK code to the assign node. This method only generates the conditional part,
+            # the assignment part will be generated in the assign node.
+            return statements, InfoStr(
+                json.dumps({"value": self.parameters.value, "mask": mask_var_name})
+            )
+
+        # This handles the normal series assignment case.
```

Comment:
Maybe this comment can clarify what normal series assignment means. Does it mean series assignment that doesn't affect parent view?